src/cycax/cycad/cycad_part.py

```diff
@@ -256,50 +282,47 @@ def insert_hole(self, hole: Holes):
 
         self.features.append(hole)
         self.move_holes.append(hole)
-
-    def side_location_calculator(self, side: str, x: float, y: float, sink: float = 0.0) -> tuple:
+        
+    def save(self):
         """
-        Calculate given the side of and object and the relative x and y location on that side where the definite location is.
-        Args:
-            x: x location
-            y: y location
-            side: side of object
-
-        Raises:
-            ValueError: if the side is not one of the available and expected sides.
+        This takes the provided part and will create its dictionary and export it to a json
+        """
+        dir_name = f"{os.getcwd()}/{self.part_no}/{self.part_no}"
+        if not os.path.exists(dir_name):
+            os.mkdir(dir_name)
+        with open(f"{dir_name}/{self.part_no}/{self.part_no}.json", "w") as jsonfile:
+            json.dump(self.export(), jsonfile, indent=4)
+            
+            
+    def export(self) -> dict:
+        """
+        This method will take the values stored within the part and export it to a dict so that it can be decoded.
 
         Returns:
-            tupel: tupel containing floats of the x, y and z values.
-
-
+            dict : The dictionary of the part.
         """
 
-        if side == TOP:
-            temp_x = x
-            temp_y = y
-            temp_z = self.z_max - sink  # Where is self.z_max defined??
-        elif side == BOTTOM:
-            temp_x = x
-            temp_y = y
-            temp_z = self.z_min + sink
-        elif side == LEFT:
-            temp_x = self.x_min + sink
-            temp_y = x
-            temp_z = y
-        elif side == RIGHT:
-            temp_x = self.x_max - sink
-            temp_y = x
-            temp_z = y
-        elif side == FRONT:
-            temp_x = x
-            temp_y = self.y_min + sink
-            temp_z = y
-        elif side == BACK:
-            temp_x = x
-            temp_y = self.y_max - sink
-            temp_z = y
-        else:
-            msg = f"Side: {side} is not one of TOP, BOTTOM, LEFT, RIGHT, FRONT, BACK."
-            raise ValueError(msg)
-
-        return temp_x, temp_y, temp_z
+        dict_metal = {
+            "name": self.poligon,
+            "type": "add",
+            "side": self.side,
+            "x": self.x,
+            "y": self.y,
+            "z": self.z,
+            "x_size": self.x_size,
+            "y_size": self.y_size,
+            "z_size": self.z_size,
+            "center": False,
+        }
+
+        dict_part = []
```

Comment:
We agreed that export returns a dict (as per the method signature) here it return a list.